src/realm/object-store/impl/object_notifier.cpp

```diff
@@ -63,22 +59,25 @@ void ObjectNotifier::run()
 {
     if (!m_table_key)
         return;
+    REALM_ASSERT(m_table);
+
+    auto it = m_info->tables.find(m_table_key.value);
+    if (it != m_info->tables.end() && it->second.deletions_contains(m_obj_key.value)) {
+        // The object was deleted in this set of changes, so report that and
+        // release all of our resources so that we don't do anything further.
+        m_change.deletions.add(0);
+        m_table = {};
+        m_table_key = {};
+        m_obj_key = {};
```

Comment:
although harmless here, this is the objkey version of the same implicit conversion, maybe worth removing this one too?

Reply:
I think I'll go ahead and remove that operator from all of the key types.